app/src/main/java/com/napzak/market/presentation/registration/RegistrationScreen.kt

```diff
@@ -0,0 +1,338 @@
+import com.napzak.market.core.designsystem.component.button.CommonButton
+import com.napzak.market.core.designsystem.component.topbar.CloseTopBar
+import com.napzak.market.core.designsystem.theme.NapzakMarketTheme
+import com.napzak.market.core.type.ProductConditionType
+import com.napzak.market.core.type.TradeType
+import com.napzak.market.presentation.registration.component.RegistrationBuyGroup
+import com.napzak.market.presentation.registration.component.RegistrationGenreButton
+import com.napzak.market.presentation.registration.component.RegistrationPhotoPicker
+import com.napzak.market.presentation.registration.component.RegistrationPlainTextField
+import com.napzak.market.presentation.registration.component.RegistrationSellGroup
+import com.napzak.market.presentation.registration.state.RegistrationUiState
+import com.napzak.market.presentation.registration.type.NumeralInputType
+import com.napzak.market.presentation.registration.type.PlainTextInputType
+import com.napzak.market.presentation.registration.type.PostFeeType
+
+@Composable
+fun RegistrationRoute(
+    tradeType: String,
+    navigateUp: () -> Unit,
+    navigateToGenreSearch: () -> Unit,
+    modifier: Modifier = Modifier,
+    viewModel: RegistrationViewModel = hiltViewModel(),
+) {
+    val uiState by viewModel.uiState.collectAsStateWithLifecycle()
+    val currentListSize = (MAX_ITEMS - uiState.imageUrlList.size).coerceAtLeast(MIN_ITEMS)
+    val getImageStorageLauncher = rememberLauncherForActivityResult(
+        ActivityResultContracts.GetMultipleContents()
+    ) { uris: List<Uri> ->
+        val updatedListSize = uiState.imageUrlList.size + uris.size
+        if (updatedListSize <= 10) {
+            viewModel.updatePhotoList(uris.map { it.toString() })
+        } else {
+            val remainingUris = uris.take(MAX_ITEMS - uiState.imageUrlList.size)
+            viewModel.updatePhotoList(remainingUris.map { it.toString() })
+        }
+    }
+    val getPhotoPickerLauncher = when (uiState.imageUrlList.size) {
+        MAX_ITEMS - 1 -> rememberLauncherForActivityResult(
+            ActivityResultContracts.PickVisualMedia()
+        ) { uri: Uri? ->
+            uri?.let { viewModel.updatePhotoList(listOf(it.toString())) }
+        }
+        else -> rememberLauncherForActivityResult(
+            ActivityResultContracts.PickMultipleVisualMedia(maxItems = currentListSize)
+        ) { uris: List<Uri> ->
+            val updatedListSize = uiState.imageUrlList.size + uris.size
+            if (updatedListSize <= 10) {
+                viewModel.updatePhotoList(uris.map { it.toString() })
+            } else {
+                val remainingUris = uris.take(MAX_ITEMS - uiState.imageUrlList.size)
+                viewModel.updatePhotoList(remainingUris.map { it.toString() })
+            }
+        }
+    }
+
+    LaunchedEffect(tradeType) {
+        viewModel.updateTradeType(if (tradeType == TradeType.SELL.label) TradeType.SELL else TradeType.BUY)
+    }
```

Comment:
```suggestion
   LaunchedEffect(Unit)
```
p2: 거래 유형은 외부로부터 전달되는 값인데 이 값이 변경될 가능성이 적다면 이렇게 해보시는 것이 조금 더 명확하지 않을까싶습니당